src/ucp/wireup/wireup_ep.c

```diff
@@ -655,6 +656,15 @@ void ucp_wireup_ep_set_next_ep(uct_ep_h uct_ep, uct_ep_h next_ep)
     ucp_proxy_ep_set_uct_ep(&wireup_ep->super, next_ep, 1);
 }
 
+void ucp_wireup_ep_unset_next_ep(ucp_wireup_ep_t *wireup_ep)
+{
+    ucs_assert(wireup_ep != NULL);
+    ucs_assert(wireup_ep->super.uct_ep != NULL);
+    wireup_ep->flags &= ~UCP_WIREUP_EP_FLAG_LOCAL_CONNECTED;
+    ucs_assert(wireup_ep->flags == 0);
+    ucp_proxy_ep_set_uct_ep(&wireup_ep->super, NULL, 0);
```

Comment:
should not we extract and destroy old one? like this:
```
uct_ep_destroy(ucp_wireup_ep_extract_next_ep());
wireup_ep->flags &= ~UCP_WIREUP_EP_FLAG_LOCAL_CONNECTED;
ucs_assert(wireup_ep->flags == 0);
```

Reply:
we do destroy the cm ep right after the call to this function - https://github.com/openucx/ucx/pull/5551/files#diff-36e34c1df2a6c773b9dda08dd8bf17eeR780

need to also set the next_ep to NULL.

Reply:
this way code looks independent but actually this does dependent things (BTW your links does not work)